lib/redis-session-store.rb

```diff
@@ -100,30 +74,45 @@ def verify_handlers!
   end
 
   def prefixed(sid)
+    return nil unless sid && sid.private_id
+    "#{default_options[:key_prefix]}#{sid.private_id}"
+  end
+
+  def prefixed_fallback(sid)
     "#{default_options[:key_prefix]}#{sid}"
   end
 
-  def session_default_values
-    [generate_sid, USE_INDIFFERENT_ACCESS ? {}.with_indifferent_access : {}]
+  def create_sid_with_empty_session(redis_connection)
+    loop do
+      sid = generate_sid
+      key = prefixed(sid)
+      if key && redis_connection.set(key, encode({}), nx: true, ex: default_redis_ttl)
+        break sid
+      end
+    end
   end
 
-  def get_session(env, sid)
-    sid && (session = load_session_from_redis(sid)) ? [sid, session] : session_default_values
-  rescue Errno::ECONNREFUSED, Redis::CannotConnectError => e
-    on_redis_down.call(e, env, sid) if on_redis_down
-    session_default_values
+  def find_session(req, sid)
```

Comment:
this is a method for the interface right? 
I know it's not necessarily standard, but what if we labeled ones that are part of the interface as `@api public`, mostly to make it easier to find (https://rubydoc.info/gems/yard/file/docs/Tags.md#api)


```suggestion
  # @api public
  def find_session(req, sid)
```

Reply:
Done!